### PR TITLE
DC Branch Support

### PR DESCRIPTION
**Pull request by darrylmelander**
Since we just added support for dc_branch elements in the Egret GMLC parser, we want to be sure Prescient includes DC branches in its logic and reporting.

### CODE REVIEW

**Comment by darrylmelander** [comment]
@bknueven I just blindly took any code that dealt with branch elements and included dc_branch elements as well. I don't know if that was always the right thing to do, and as a result this PR is marked as WIP. Feel free to take off the WIP status after you've reviewed it if you feel OK about where I added DC support.

Most of the changes boiled down to looping over two sets of elements instead of just one. Wherever there used to be for loops over branch elements, I leaned toward including both kinds of branches in a single loop rather than copying the code into a second loop. Also, I added a couple of TODO comments I hope you'll notice and resolve.

**Review by bknueven — COMMENTED** [review]
Just a few minor comments

File: prescient/engine/egret/egret_plugin.py [hunk 1]
```diff
@@ -676,13 +676,18 @@ def _ensure_contingencies_monitored(options:Options, md:EgretModel, initial_ruc:
...
+                    f"DC Branch {bn} has `in_service`:False")
     for bn, b in md.elements('bus'): 
         if not b.get('in_service', True):
             raise RuntimeError(f"Buses cannot be removed from service in Prescient")
 
     if options.monitor_all_contingencies:
         key = []
-        for bn, b in md.elements('branch'):
+        for bn, b in (kv for btype in ('branch', 'dc_branch')
+                         for kv in md.elements(btype)):
```
Comment (bknueven) [inline prescient/engine/egret/egret_plugin.py:690]:
```suggestion
        for bn, b in md.elements('branch'):
```
We don't need these here either.

File: prescient/engine/egret/egret_plugin.py [hunk 2]
```diff
@@ -692,6 +697,7 @@ def _ensure_contingencies_monitored(options:Options, md:EgretModel, initial_ruc:
         key = tuple(key)
         if key not in _ensure_contingencies_monitored.contingency_dicts:
             mapping_bus_to_idx = { k : i for i,k in enumerate(md.data['elements']['bus'].keys())}
+            # TODO: Should we include DC branches here?
```
Comment (bknueven) [inline prescient/engine/egret/egret_plugin.py:700]:
```suggestion
```
No -- it's complicated.

File: prescient/engine/egret/ptdf_manager.py [hunk 3]
```diff
@@ -69,7 +69,8 @@ def update_active(self, md):
             self._active_interface_constraints[i_n] += 1
 
         misses = 0
-        for bn, branch in md.elements(element_type='branch'):
+        for bn, branch in (kv for btype in ('branch', 'dc_branch') 
+                              for kv in md.elements(element_type=btype)):
```
Comment (bknueven) [inline prescient/engine/egret/ptdf_manager.py:73]:
```suggestion
        for bn, branch in md.elements(element_type='branch'):
```

**Pull request merged**